x-pack/plugins/enterprise_search/public/applications/app_search/components/crawler/crawler_single_domain_logic.ts

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { kea, MakeLogicType } from 'kea';
+
+import { flashAPIErrors } from '../../../shared/flash_messages';
+
+import { HttpLogic } from '../../../shared/http';
+import { EngineLogic } from '../engine';
+
+import { CrawlerDomain } from './types';
+import { crawlerDomainServerToClient } from './utils';
+
+export interface CrawlerSingleDomainValues {
+  dataLoading: boolean;
+  domain: CrawlerDomain | null;
+}
+
+interface CrawlerSingleDomainActions {
+  fetchDomainData(domainId: string): { domainId: string };
+  onReceiveDomainData(domain: CrawlerDomain): { domain: CrawlerDomain };
+}
+
+export const CrawlerSingleDomainLogic = kea<
+  MakeLogicType<CrawlerSingleDomainValues, CrawlerSingleDomainActions>
+>({
+  path: ['enterprise_search', 'app_search', 'crawler', 'crawler_single_domain'],
+  actions: {
+    fetchDomainData: (domainId) => ({ domainId }),
+    onReceiveDomainData: (domain) => ({ domain }),
+  },
+  reducers: {
+    dataLoading: [
+      true,
+      {
+        onReceiveDomainData: () => false,
+      },
+    ],
+    domain: [
+      null,
+      {
+        onReceiveDomainData: (_, { domain }) => domain,
+      },
+    ],
+  },
+  listeners: ({ actions }) => ({
+    fetchDomainData: async ({ domainId }) => {
+      const { http } = HttpLogic.values;
+      const { engineName } = EngineLogic.values;
+
+      try {
+        const response = await http.get(
+          `/api/app_search/engines/${engineName}/crawler/domains/${domainId}`
```

Comment:
In this context, this should be OK but I would suspect that with analytics for instance we would want/need to escape the dynamic parts of the URL path.

Reply:
Can you elaborate on this? I'm not sure I understand why we would want/need to escape the dynamic parts in the context of analytics.